pkg/querier/blocks_store_queryable.go

```diff
@@ -590,6 +590,9 @@ func (q *blocksStoreQuerier) fetchSeriesFromStores(
 			// TODO(goutham): we should ideally be passing the hints down to the storage layer
 			// and let the TSDB return us data with no chunks as in prometheus#8050.
 			// But this is an acceptable workaround for now.
+
+			// Only fail the function if we have validation error. We should return blocks that were successfully
```

Comment:
We should also fail the function when store-gateways return a limit error. It'll be good to avoid unnecessary retries in that case.